src/test/scala/za/co/absa/commons/reflect/ReflectionUtilsSpec.scala

```diff
@@ -136,6 +136,14 @@ class ReflectionUtilsSpec extends AnyFlatSpec with Matchers with MockitoSugar {
     ReflectionUtils.extractFieldValue[Boolean](Foo, "x") should be(Foo.x)
   }
 
+  // this class must be outside the test to trigger the exception
+  case class WrappingClass(x: ClassWithCyclicAnnotation)
```

Comment:
Minor: another such dummy classes are located in the test companion object below. I would move this one there as well for the sake of consistency.